connman/src/technology.c

```diff
@@ -1946,6 +1982,17 @@ bool __connman_technology_enable_from_config()
 
 	keyfile = __connman_storage_load_global();
 	if (!keyfile) {
+		/*
+		 * When the settings file does not exist create it similarly to
+		 * technology is initialization. Since this concerns new
+		 * users set the global offlinemode to "false" as new users
+		 * should not have offline mode set.
+		 */
+		if (global_offlinemode)
```

Comment:
`_set_offlinemode()` already does this check, so you could just call it directly here. Also, I'd prefer using `false` instead of the variable. (one less thing to remember)